lessons/2_more_strings_and_functions/lesson.py

```diff
@@ -372,25 +419,85 @@ def inner_func():
 # 1. Write a function that takes a string, sorts the characters in the string, and returns a string
 #    with the individual characters sorted. Name this function sort_chars.
 
-# 2. Write a function that takes a string and returns the value of the character at the index which is half of the
-#    total length of the string  if the string is even, the value of the penultimate character if the string is odd, 
+def sort_chars(string):
+    string_sorted = sorted(string)
+    string_sorted = "".join(string_sorted)
+    return string_sorted
+
+print(my_random_chars, "sorted is", sort_chars(my_random_chars))
+
+# # i realized this literally returns e.g. 'the total length of the string', 'l;k2;34lksd;lkfg', 'is', 16
+# # how do i have it return e.g. the total length of the string l;k2;34lksd;lkfg is 16
+
+# # # this seems to be why it is failing test.py
+# # # i think i fixed it, by just pulling the commentary out of the function and into the print.
+
+# 2. Write a function that takes a string and returns the value of the character at the index which is
+#    half of the total length of the string  if the string is even,
+#    the value of the penultimate character if the string is odd, 
 #    and 'Dobby' if an empty-string or length 1. Name this function dobby_search.
 
+def dobby_search(string):
+    if (len(string) % 2) == 0:
+        return string[len(string)//2]
+    else:
+        return string[len(string)-1]
+
+print(dobby_search(my_random_chars))
+
+# # # this one is failing the test, but i am not sure why.
```

Comment:
It's because it needs to return 'Dobby' for emtpy strings and strings of length 1!